example/src/components/Stats.tsx

```diff
@@ -0,0 +1,188 @@
+import { LineChart } from 'react-native-charts-wrapper';
+
+import { Typo } from './Typo';
+
+type StatsProp = {
+  stats: any[];
+  label: string;
+};
+
+const notPlottableStats = [
+  'kind',
+  'rid',
+  'packetsLost',
+  'packetsReceived',
+  'bytesReceived',
+  'framesReceived',
+  'framesDropped',
+  'bytesSent',
+  'packetsSent',
+  'framesEncoded',
+];
+
+const chartDescriptionConfig = {
+  text: '',
+  textSize: 0,
+};
+
+const yAxisConfig = {
+  left: {
+    enabled: false,
+  },
+};
+
+export const Stats = ({ stats, label }: StatsProp) => {
+  const getValues = useCallback(
+    (label: string, chart: string) => {
+      const timestampsWithLabel = stats.filter((obj) => {
+        return Object.keys(obj).includes(label);
+      });
+
+      const values = timestampsWithLabel.map((obj) => {
+        return { y: obj[label][chart] !== null ? obj[label][chart] : 0 };
+      });
+
+      return values;
+    },
+    [stats]
+  );
+
+  const generateLimitationDurationInitData = useCallback(
+    (name: string, color: string) => {
+      return {
+        label: name,
+        values: [{ y: 0 }],
+        config: {
+          drawValues: false,
+          drawCircles: false,
+          color: processColor(color),
+          lineWidth: 2,
+        },
+      };
+    },
+    []
+  );
+
+  const getLimitationDurationsDataset = useCallback(
+    (label: string, chart: string) => {
+      const timestampsWithLabel = stats.filter((obj) => {
+        return Object.keys(obj).includes(label);
+      });
+
+      const combinedValues = timestampsWithLabel.map((obj) => {
+        return obj[label][chart] !== null ? obj[label][chart] : 0;
+      });
+
+      const res = [
+        generateLimitationDurationInitData('bandwidth', 'blue'),
+        generateLimitationDurationInitData('cpu', 'red'),
+        generateLimitationDurationInitData('none', 'green'),
+        generateLimitationDurationInitData('other', 'yellow'),
+      ];
+
+      combinedValues.forEach((obj) => {
+        res[0].values.push({ y: obj['bandwidth'] });
+        res[1].values.push({ y: obj['cpu'] });
+        res[2].values.push({ y: obj['none'] });
+        res[3].values.push({ y: obj['other'] });
+      });
+
+      return res;
+    },
+    [stats]
+  );
+
+  const getLineChart = useCallback((dataSets, isMultiValue: boolean) => {
+    return (
+      <LineChart
+        style={styles.chart}
+        data={{
+          dataSets,
+        }}
+        yAxis={yAxisConfig}
+        xAxis={{
+          drawLabels: false,
+        }}
+        chartDescription={chartDescriptionConfig}
+        legend={{ enabled: isMultiValue, drawInside: isMultiValue }}
+        marker={{ enabled: false }}
+        logEnabled={false}
+        touchEnabled={false}
+      />
+    );
+  }, []);
+
+  return (
+    <View>
+      <Collapse>
+        <CollapseHeader>
+          <Typo variant="h5">{label}</Typo>
+        </CollapseHeader>
+        <CollapseBody>
+          <View style={styles.label}>
```

Comment:
Probably the charts are still rendering even though they're collapsed. Could you not render them if they're collapsed?

Reply:
Unfortunately no, the only idea I have is to use onToggle prop that knows whether the body is collapsed and then do some conditional rendering, but the downside is that as the name suggests this is only called when an element is pressed, so if by default everything is collapsed it will still be rendered until somebody clicks on it twice.